runtime/src/lib.rs

```diff
@@ -376,6 +189,18 @@ pub fn handle_mailbox_commands(drivers: &mut Drivers) -> CaliptraResult<()> {
     }
     loop {
         enter_idle(drivers);
+        if drivers.is_shutdown {
+            cprintln!("[rt] is_shutdown");
+            caliptra_drivers::report_fw_error_non_fatal(CaliptraError::RUNTIME_SHUTDOWN.into());
+            caliptra_drivers::report_fw_error_fatal(CaliptraError::RUNTIME_SHUTDOWN.into());
+            loop {
+                // SoC firmware might be stuck waiting for Caliptra to finish
+                // executing this pending mailbox transaction. Notify them that
+                // we've failed.
+                unsafe { caliptra_drivers::Mailbox::abort_pending_soc_to_uc_transactions() };
+            }
+        }
```

Comment:
I think this needs to be inside the `is_cmd_ready` check. Otherwise we'll do this any time we wake up from any interrupt.

Reply:
We can just leverage the fatal error flow  - propagate it as an error up to main.